kafka/src/main/java/io/micronaut/configuration/kafka/processor/ConsumerStateSingle.java

```diff
@@ -132,13 +132,14 @@ private boolean resolveWithErrorStrategy(@Nullable ConsumerRecords<?, ?> consume
                 // Check how many retries so far
                 final int currentRetryCount = getCurrentRetryCount(consumerRecord);
                 if (info.retryCount >= currentRetryCount) {
+                    // Move back to the previous position
+                    kafkaConsumer.seek(topicPartition, consumerRecord.offset());
 
-                    // We will retry this batch again next time
+                    // We will retry this batch again next time unless the exception handler chooses not to
                     if (info.shouldHandleAllExceptions) {
                         handleException(e, consumerRecords, consumerRecord);
                     }
-                    // Move back to the previous position
-                    kafkaConsumer.seek(topicPartition, consumerRecord.offset());
```

Comment:
The one problem that comes to mind here is that it is a potentially breaking change in behavior.

i.e., will this break existing ExceptionHandlers that assume the current offset position when the handler is invoked has not already been moved back?